DynamicFlightStorageSimulation/SimulationEventHandlerFactory.cs

```diff
@@ -14,7 +14,7 @@ public static AsyncEventHandler<BasicDeliverEventArgs> GetEventHandler<TMessage>
         {
             return async (sender, ea) =>
             {
-                if (channel is null)
+                if (channel is null ) // might need "|| channel.IsClosed"?
```

Comment:
The issue you ran into here also needs to be fixed. Possibly in another PR though.

The issue stems from RabbitMQ disconnecting you (possibly because you were in a debugger and not responding to it's pings). When that happens we first or all need to call "reconnect" for it to not complain. The problem is just that our queues are created with the auto-delete flag so RabbitMQ has now also deleted the queues we're trying to read from. 

I don't know a great solution to that problem though. Maybe it's best just to throw and break the app. Annoying for debugging though...